packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -100,41 +101,75 @@ class _SnapshotListTitle extends StatelessWidget {
   }
 }
 
-class _SnapshotListItems extends StatelessWidget {
-  _SnapshotListItems({Key? key, required this.controller}) : super(key: key);
+class _SnapshotListItems extends StatefulWidget {
+  const _SnapshotListItems({Key? key, required this.controller})
+      : super(key: key);
 
   final DiffPaneController controller;
-  final headerHeight = 1.20 * defaultRowHeight;
+
+  @override
+  State<_SnapshotListItems> createState() => _SnapshotListItemsState();
+}
+
+class _SnapshotListItemsState extends State<_SnapshotListItems>
+    with AutoDisposeMixin {
+  final _headerHeight = 1.20 * defaultRowHeight;
+  late ScrollController _scrollController;
+  late int _currentSnapshotsLength;
+
+  @override
+  void initState() {
+    _scrollController = ScrollController();
+    super.initState();
```

Comment:
call to super initState should always be first. Opposite of dispose, where call to super should always be last.